### DISCUSSION TITLE
Regarding the YoloV8 traning with roloflow dataset

### DISCUSSION THREAD

**Discussion by subhakarkoduru** [ultralytics/ultralytics#10091] [Q&A]
I had a small dataset which i had annotated using roboflow and used that dataset for custom training yolov8 model, Now i want to add few images along with thier labels(I don't need annotation for them i already have labels for them) to dataset before giving them to yolov8 model training. So i had downloded the dataset in github and added my images and labels to the dataset and downloading it directly in my colab for traning. Is it feasabile or can anybody please sugest me more effiecient method to do it.

**Comment by pderrenger** [comment]
@subhakarkoduru hey there! 🌟 Adding new images and their corresponding labels directly to your dataset on GitHub, and then downloading it for training in Colab, is indeed a feasible method. If you've already annotated them and have the labels ready, you're on the right track.

For a slightly more streamlined approach, you might consider:
1. Ensuring your added images and labels follow the same format as the existing dataset.
2. After adding the images and labels, just double-check the dataset directory structure to make sure everything is correctly placed.

Here's a small example on how you might load and add data to your existing dataset in Colab:

```python
# Clone your dataset repository
!git clone https://github.com/yourusername/yourdatasetrepo.git

# Optionally, copy new images and labels into the cloned repo directory
# Example: !cp path_to_new_images/* yourdatasetrepo/images/
#          !cp path_to_new_labels/* yourdatasetrepo/labels/

# Now, proceed with your YOLOv8 training
# Remember to adjust paths based on your dataset structure
```

Remember, the key is keeping your dataset organized and in the correct format expected by YOLOv8 (images in one directory, labels in another, matching file names). This method ensures that your workflow remains flexible and manageable. Good luck with your training! 🚀
